docs/SUMMARIZER_QUAY_IMAGE_CLEANUP.md

```diff
@@ -0,0 +1,358 @@
+- ✅ Keeping the `latest` tag (always protected)
+- ✅ Keeping all tags ending with `-release` (always protected)
+- ✅ Keeping user-specified protected tags (when running manually)
+- ✅ Keeping all images created within the retention period
+- ✅ Showing age and creation date for each tag processed
+- ✅ Verifying deletions by checking if tags still exist after deletion
+- ✅ Tracking and reporting failed deletions with detailed error messages
+- ✅ Displaying a summary list of all failed tags for easy troubleshooting
+
+### Required Secrets
+
+The workflow uses existing GitHub secrets (already configured for your build workflow):
+- `QUAY_USERNAME`: Quay.io username
+- `QUAY_PASSWORD`: Quay.io password/token
+
+### Images Managed
+
+The workflow processes these Summarizer container images:
+- `quay.io/ecosystem-appeng/aiobs-metrics-ui`
+- `quay.io/ecosystem-appeng/aiobs-metrics-alerting`
+- `quay.io/ecosystem-appeng/aiobs-mcp-server`
+
+## Usage Examples
+
+### 1. Test with Dry Run (Recommended First)
+
+Before running cleanup for real, test with dry run:
+
+```bash
+# Via GitHub UI:
+Actions → Cleanup Old Summarizer Container Images → Run workflow
+- Dry run: ✓ (checked)
+- Retention days: 30
+```
+
+This will show what would be deleted without actually deleting anything.
+
+### 2. Cleanup with Custom Retention
+
+To keep images for 60 days instead of 30:
+
+```bash
+# Via GitHub UI:
+Actions → Cleanup Old Summarizer Container Images → Run workflow
+- Dry run: ☐ (unchecked)
+- Retention days: 60
+```
+
+### 3. Aggressive Cleanup (Keep Only Recent)
+
+To keep only images from the last 7 days:
+
+```bash
+# Via GitHub UI:
+Actions → Cleanup Old Summarizer Container Images → Run workflow
+- Dry run: ☐ (unchecked)
+- Retention days: 7
+- Protected tags: (leave empty)
+```
+
+### 4. Protect Specific Tags During Cleanup
+
+To protect specific tags in addition to the automatic protections:
+
+```bash
+# Via GitHub UI:
+Actions → Cleanup Old Summarizer Container Images → Run workflow
+- Dry run: ☐ (unchecked)
+- Retention days: 30
+- Protected tags: 1.5.0,2.0.0-beta,hotfix-123
+```
+
+This will protect:
+- `latest` (always)
+- `*-release` (always)
+- `1.5.0`, `2.0.0-beta`, `hotfix-123` (user-specified)
+
+## Workflow Output
+
+The workflow provides detailed output for each image:
+
+```
+🔍 Processing image: quay.io/ecosystem-appeng/aiobs-metrics-ui
+📅 Retention policy: Keep images from last 30 days
+🔒 User-protected tags: 1.5.0,hotfix-123
+
+Found 45 tags total
+
+🔒 Protecting tag: latest (reserved tag)
+🔒 Protecting tag: 1.0.0-release (release tag)
+🔒 Protecting tag: 1.5.0 (user-specified)
+🔒 Protecting tag: hotfix-123 (user-specified)
+✅ Keeping tag: 1.0.1 (age: 5 days, created: 2025-11-27T10:30:00Z)
+🗑️  Deleting tag: 0.9.8 (age: 45 days, created: 2025-10-18T08:15:00Z)
+   ✅ Tag '0.9.8': Deleted successfully and verified
+🗑️  Deleting tag: 0.9.7 (age: 50 days, created: 2025-10-13T08:15:00Z)
```

Comment:
not sure we need this whole document ?

Reply:
Are you suggesting I remove this doc that shows how the cleanup workflow works?